pkg/artifactory/resource_artifactory_virtual_repository.go

```diff
@@ -186,6 +191,11 @@ func resourceVirtualRepositoryDelete(d *schema.ResourceData, m interface{}) erro
 	repo := unpackVirtualRepository(d)
 
 	resp, err := c.V1.Repositories.DeleteVirtual(context.Background(), *repo.Key)
+
+	if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.